### PR TITLE
Feature/refresh token

### PR DESCRIPTION
**Pull request by yevhen-turovets**
The task in Trello https://trello.com/c/E5IdSUQv/31-create-refreshtoken-entity

### CODE REVIEW

File: api/app/Entity/JWTRefreshToken.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+class JWTRefreshToken extends Model
+{
+    protected $fillable = [
+        'user_id', 'token',
+    ];
+
+    public function user(): BelongsTo
+    {
+        return $this->belongsTo(User::class, 'user_id');
+    }
```
Comment (aLLeXUs) [inline api/app/Entity/JWTRefreshToken.php:17]: The reverse relation should be added to the `User` entity.

File: api/app/Repositories/JWTRefreshToken/JWTRefreshTokenRepository.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+{
+    public function getById(int $id): ?JWTRefreshToken
+    {
+        return JWTRefreshToken::find($id);
+    }
+
+    public function getByUserId(int $userId): ?JWTRefreshToken
+    {
+        return JWTRefreshToken::firstWhere('user_id', $userId);
+    }
```
Comment (aLLeXUs) [inline api/app/Repositories/JWTRefreshToken/JWTRefreshTokenRepository.php:20]: Users can have more than one refresh token, so there is no sense to get only first. I think we can remove this for now.

File: api/database/migrations/2020_08_05_135254_create_jwt_refresh_tokens_table.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('jwt_refresh_tokens', function (Blueprint $table) {
+            $table->id();
+            $table->timestamps();
+            $table->string('token');
+            $table->unsignedBigInteger('user_id');
+
```
Comment (aLLeXUs) [inline api/database/migrations/2020_08_05_135254_create_jwt_refresh_tokens_table.php:21]: I forgot to mention that an expiration date should be present here. I have updated the Trello ticket, please, add it too.

File: api/app/Repositories/JWTRefreshToken/JWTRefreshTokenRepository.php (hunk 2 above)
Comment (yevhen-turovets) [reply api/app/Repositories/JWTRefreshToken/JWTRefreshTokenRepository.php:20]: Change firstWhere on Where or remove this method?

Comment (aLLeXUs) [reply api/app/Repositories/JWTRefreshToken/JWTRefreshTokenRepository.php:20]: @yevhen-turovets remove the method.

File: api/app/Entity/User.php [hunk 4]
```diff
@@ -38,6 +39,11 @@ class User extends Authenticatable implements JWTSubject
         'email_verified_at' => 'datetime',
     ];
 
+    public function refreshToken(): HasMany
```
Comment (aLLeXUs) [inline api/app/Entity/User.php:42]: It relates to many records, so we should name it `refreshTokens`.

File: api/app/Repositories/JWTRefreshToken/JWTRefreshTokenRepositoryInterface.php [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+declare(strict_types=1);
+
+namespace App\Repositories\JWTRefreshToken;
+
+use App\Entity\JWTRefreshToken;
+
+interface JWTRefreshTokenRepositoryInterface
+{
+    public function getById(int $id): ?JWTRefreshToken;
+    public function getByUserId(int $userId): ?JWTRefreshToken;
```
Comment (aLLeXUs) [inline api/app/Repositories/JWTRefreshToken/JWTRefreshTokenRepositoryInterface.php:12]: The method must be removed from the interface too.

**Pull request merged**